marge/bot.py

```diff
@@ -72,8 +72,8 @@ def _run(self, repo_manager):
                 return
 
             big_sleep = max(0,
-                            min_time_to_sleep_after_iterating_all_projects_in_secs -
-                            time_to_sleep_between_projects_in_secs * len(projects))
+                            min_time_to_sleep_after_iterating_all_projects_in_secs
+                            - time_to_sleep_between_projects_in_secs * len(projects))
```

Comment:
Ugh, this is a horrible change. :slightly_frowning_face: I guess it's `pylint` forcing us to move the - operator to the beginning of the line there?

Reply:
Yep:
- https://www.flake8rules.com/rules/W503.html
- https://www.flake8rules.com/rules/W504.html

Reply:
Their reasoning is not completely off IMO: https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

Reply:
Thanks, I looked at that PEP now and briefly in the _The TeXBook_ by Donald Knuth that it refers to. I think that the argument (from the Python community) is a bit weak in this case, to be honest. While formulating advanced equations in a TeX publication (the original Knuth use case, p.195) and programming shares _some_ similarity, I think van Rossum stretches it too far in this case. Breaking with the age-old tradition comes at a price as well.

Let me be clear. I _do_ agree that the Python style keeps the operands closer to the `+`, `-` etc operand, but at least to me, the example given there (with `gross_wages`, `taxable_interest` etc) is perhaps not complex enough to make it worth it. Both examples are still quite readable, in this particular example.

Also:

> For decades the recommended style was to break after binary operators.

...I think there are _reasons_ for this, and that reason is simply **not** just that everybody else are stupid^Wless enlightened people. :grin: 

----

Having all that said, this is a Python project and this is the BDFL's suggested way to write Python code. Let's go with it. :+1:

Reply:
Btw, the PEP actually states this:

> In Python code, it is permissible to break before or after a binary operator, as long as the convention is consistent locally. For new code Knuth’s style is suggested.

...so it would be fine to disable either W503 or W504, depending on e.g. which one is least work converting existing code. I think the key which the PEP encourages is internal consistency.